gobblin-core/src/main/java/gobblin/source/extractor/hadoop/RawByteExtractor.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.source.extractor.hadoop;
+
+import gobblin.source.extractor.DataRecordException;
+import gobblin.source.extractor.Extractor;
+import gobblin.source.extractor.utils.BufferedRawByteIterator;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+
+/**
+ * Implementation of {@link gobblin.source.extractor.Extractor} that reads a file as a sequence of byte arrays.
+ * Uses a {@link gobblin.source.extractor.utils.BufferedRawByteIterator}.
+ */
+public class RawByteExtractor implements Extractor<String, byte[]> {
```

Comment:
We should consider using ByteBuffers (direct ByteBuffers in particular) to avoid repeated re-allocation of large byte arrays, 0 initialization and the cost of moving those during GC.